packages/graphql/src/schema/create-relationship-fields/index.ts

```diff
@@ -777,25 +774,29 @@ function createRelationshipFields({
         });
 
         if (!rel.writeonly) {
-            const nodeFieldsBaseArgs = {
-                where: `${rel.typeMeta.name}Where`,
-                options: `${rel.typeMeta.name}Options`,
+            const relationshipField: { type: string; description?: string; directives: Directive[]; args?: any } = {
+                type: rel.typeMeta.pretty,
+                description: rel.description,
+                directives: graphqlDirectivesToCompose(rel.otherDirectives),
             };
 
-            const nodeFieldsArgs = addDirectedArgument(nodeFieldsBaseArgs, rel);
+            // Subgraph schemas do not support arguments on relationship fields
+            if (!(subgraph && !rel.typeMeta.array)) {
```

Comment:
Maybe moving this to a separate variable could provide more context for this logic?